### ISSUE TITLE
Decameron Website!

### ISSUE THREAD

**Issue by laurenmcguigan** [jlm323/DecameronProject#22]
@ebeshero I'm working on the xslt right now for our website and just had a few questions. 
1. There is a tag called `<front> </front>` what is that being used to tag? Is it TEI? Should I just remove it for the HTML.
2. When I am doing the XSLT, if there is something coded that I don't need for the HTML is there a way for me to remove it? And if I want to keep some markup like `<persName>`  and `<placeName>` Do I need to add it to the XSLT, or will it automatically transfer over?

**Comment by laurenmcguigan** [comment]
@ebeshero Also, I can't really check my output because I still have that problem because I have a Mac. Remember from last time?  @spadafour Are you on campus at all today to help me fix this? Or can you send some instructions?

**Comment by ebeshero** [comment]
@laurenmcguigan  Yes, you're right about the `<front>` tag: it's part of the TEI structure, and won't be part of HTML. So here's some basic outer structure of TEI to be aware of:

```
<TEI>
     <teiHeader>
                [....lots of stuff here, metadata: You might want to output a little bit of this in the `<body>` of your HTML and maybe someplace small at the bottom of your page, but it's up to you...]
     </teiHeader>
     <text>
             <front>.... </front>
            <body>....</body>
           <back>....</back>
      </text>
</TEI>
```

Anything that's inside `<text>` is somehow part of the body of the TEI file, but I can't remember what we set up in the `<front>` portion: It might be a prologue to the Decameron? You'll want to survey the contents. Sometimes the `<front>` can have heading or introductory information in it. 

HTML structure is a lot simpler, just for presentation view, and you can pull from ANY parts of the TEI you want and **remix them** as you need to. (There's no reason you can't output stuff from the TEI header down at the bottom of your page, for example.) 

To suppress an element from being output in the TEI, you have a couple of options:
1) When you write your template match on the document node (the one that matches on '/') and converts TEI to HTML, where you write up the basic structure of your HTML file,  inside that you'll have an `<xsl:apply-templates>`. Use the `@select` attribute to select ONLY the portion of the TEI file you want to output in your HTML.

2) Later in the document, say you're working on the body `<p>` elements, and you don't want to output the contents of a particular element. You can always write a template that matches on an element, and make it empty, with NOTHING inside (NO xs:apply-templates or anything). That's a way to suppress a particular node. 

i think you want to do the first of these, right? Not the second so much because probably the paragraph level you're preserving the tagging you've applied, outputting it in <span> elements for the HTML...

Also, @spadafour posted his notes on what to do with your XSLT output in the DH-ClassHub, so you can search for that in the Issues page there--you might find it more quickly than I do as I'm multitasking right now...

**Comment by ebeshero** [comment]
@laurenmcguigan @spadafour Was it this?
https://github.com/ebeshero/DHClass-Hub/issues/157

Or if you can't get hold of Rob, save your output file and push it to the Troubleshooting folder in the DHClass-Hub, and I'll pull it in from there and see if I can figure out what's happening. Is it to do with your file extension? Make sure you save your output file as 
**something.html** 
And then you should be able to open and look at it in a web browser. Let's get some documentation posted so we always know how to solve this problem (whatever it is?) for Mac people in future!

**Comment by laurenmcguigan** [comment]
@ebeshero So this is what I have right now

```
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
    xmlns:xs="http://www.w3.org/2001/XMLSchema"
    xpath-default-namespace="http://www.tei-c.org/ns/1.0"
    xmlns="http://www.w3.org/1999/xhtml"
    exclude-result-prefixes="xs"
    version="2.0">

    <xsl:output method="xhtml" encoding="utf-8" doctype-system="about:legacy-compat"
        omit-xml-declaration="yes"/>

    <xsl:template match="/">
        <html>
            <head><title>Decameron</title></head>
            <body>

                <h1>The Decameron</h1>
                <h2>Table of Contents</h2>
                <ul>
                    <xsl:apply-templates select="//text" mode="toc"/>
                </ul>
                <hr/>

                <div id="main">
                    <xsl:apply-templates select="//text"/>

                </div>

            </body>

        </html>

    </xsl:template>

    <xsl:template match="body">
        <xsl:apply-templates/>
    </xsl:template>

    <xsl:template match="p">
        <p n="{count(preceding::p) + 1}">
            <xsl:apply-templates/>
        </p>
    </xsl:template>

    <xsl:template match="div[@type='Day']//head">
        <h2>
            <xsl:apply-templates/>
        </h2>
    </xsl:template>

    <xsl:template match="div[@type='Day']//head" mode="toc">
        <li>
            <xsl:apply-templates/>
        </li>
    </xsl:template>

    <xsl:template match="div[@type='Day']//head">
        <h1>
            <xsl:apply-templates/>
        </h1>
    </xsl:template>

    <xsl:template match="div[@type='novella']//head">
        <h2>
            <xsl:apply-templates/>
        </h2>
    </xsl:template>

    <xsl:template match="div[@type='introduction']//head">
        <h2>
            <xsl:apply-templates/>
        </h2>
    </xsl:template>


</xsl:stylesheet>
```

What I want to do is make a table of contents with all the days and then within the text, have each day be a h1 element and each novella within be an h2 element. Can I do this twice like a did above to make a table of contents made of h1s and then also have h1s within the text? Also it seems like my output is only grabbing the first day and the first introduction. Why is that? Do I need to grab each day individually?

**Comment by laurenmcguigan** [comment]
Also, I have //head in there, so I don't know why it is grabbing the entire div element for the TOC

**Comment by ebeshero** [comment]
@laurenmcguigan Let me take a look at your code, and even better, can you pull in any changes on your GitHub from upstream master, and practice **pushing** your XSLT file into GitHub, and doing a **pull request** on Jess's GitHub on the web, so then I can just pull it in and check it locally? You'll want to make sure you're comfortable with a process for doing that when Becca and I are not around, so this is a great moment to practice. 

Here's [a simple list of steps](https://github.com/jlm323/DecameronProject/issues/4): https://github.com/jlm323/DecameronProject/issues/4

**Comment by ebeshero** [comment]
Believe it or not, when you work on a project and have lots of files going back and forth, you actually memorize it like a routine! Ideally, you get a routine for GitHub down in your fingers. :-) And if you have your file in the GitHub repo, I can write comments on specific lines right there--it'll be easier to see live. @laurenmcguigan

**Comment by laurenmcguigan** [comment]
@ebeshero It is uploaded on the Decameron site. It's called readDecam.xsl

**Comment by ebeshero** [comment]
Yay! Thank you @laurenmcguigan  and perfect timing. I'll pull it in and open it up here and run it and post line-comments for you.

**Comment by laurenmcguigan** [comment]
@ebeshero Okay, quick question. I was able to output Day One with h1 headers and h3 headers for the novellas. Yay!   However, the only thing that is outputting is the elements I added to the XSLT for example div @type='Day' @corresp="day01" And the same for the novellas. So, things like persName and placeName are not outputting. Do I need to add them to the XSLT to get them to output, or is there a simpler way of getting the rest of the code to output with the original elements?

**Comment by ebeshero** [comment]
@laurenmcguigan You need a special template match on "persName" and another one on "placeName", since without special instructions, XSLT will just apply its default rule (spit out text contents) to all the TEI elements you don't specify or screen out from your output.

**Comment by RJP43** [comment]
@laurenmcguigan basically you have to tell XSLT to grab them too like how you did before but you are gonna want to wrap them in an HTML tag. It all depends on the exacts of what you want to do with them. Maybe do span elements with a class for whether they are person or place then you can make a certain style in your css for each to call on the group of them.

**Comment by ebeshero** [comment]
@laurenmcguigan I just pushed in a couple of repairs to your XSLT...I'll add comments to highlight what I changed.

**Comment by laurenmcguigan** [comment]
THANK YOU! @ebeshero @RJP43 ! I will work a little more on each day, and then upload all the HTML Days to github so we can add them to the pages I made last week!

**Comment by laurenmcguigan** [comment]
@RJP43 so, should I change `<persName>` to `<span class="person">` ? Does that work?

**Comment by ebeshero** [comment]
@laurenmcguigan Here: you'll find this helpful in deciding about what to output, I think: http://dh.obdurodon.org/class-and-span.html
